File: GXpert/GXpert.Web/Modules/ServerTypes/Attendance/RaiseHandLiveSessionColumns.ts
import { ColumnsBase, fieldsProxy } from "@serenity-is/corelib";
import { Column } from "@serenity-is/sleekgrid";
import { RaiseHandLiveSessionRow } from "./RaiseHandLiveSessionRow";

export interface RaiseHandLiveSessionColumns {
    Id: Column<RaiseHandLiveSessionRow>;
    LiveSessionlogId: Column<RaiseHandLiveSessionRow>;
    StudentPrn: Column<RaiseHandLiveSessionRow>;
    SessionTime: Column<RaiseHandLiveSessionRow>;
    HandRaiseStatus: Column<RaiseHandLiveSessionRow>;
    PlayListTitle: Column<RaiseHandLiveSessionRow>;
    ActivationDeviceId: Column<RaiseHandLiveSessionRow>;
}

export class RaiseHandLiveSessionColumns extends ColumnsBase<RaiseHandLiveSessionRow> {
    static readonly columnsKey = 'Attendance.RaiseHandLiveSession';
    static readonly Fields = fieldsProxy<RaiseHandLiveSessionColumns>();
}

File: GXpert/GXpert.Web/Modules/Attendance/RaiseHandLiveSession/RaiseHandLiveSessionColumns.cs
using Serenity.ComponentModel;
using System;
using System.ComponentModel;

namespace GXpert.Attendance.Columns;

[ColumnsScript("Attendance.RaiseHandLiveSession")]
[BasedOnRow(typeof(RaiseHandLiveSessionRow), CheckNames = true)]
public class RaiseHandLiveSessionColumns
{
    [EditLink, DisplayName("Db.Shared.RecordId"), AlignRight]
    public int Id { get; set; }
    public int LiveSessionlogId { get; set; }
    public string StudentPrn { get; set; }
    public DateTime SessionTime { get; set; }
    public short HandRaiseStatus { get; set; }
    public string PlayListTitle { get; set; }
    public string ActivationDeviceId { get; set; }
}